client/components/phone-input/index.jsx

```diff
@@ -48,19 +55,14 @@ const PhoneInput = React.createClass( {
 		};
 	},
 
-	componentWillReceiveProps( nextProps ) {
-		if ( this.props.value !== nextProps.value && nextProps.value ) {
-			const country = this.guessCountryFromValue( nextProps.value );
-			if ( country && country.isoCode !== this.getCountry().isoCode ) {
-				this.props.onChange( { countryCode: country.isoCode, value: nextProps.value } );
-			}
-		}
+	componentWillReceiveProps() {
+		// todo ensure the number is formatted if when loading from somewhere else
 	},
 
 	componentWillUpdate( nextProps ) {
-		const currentFormat = this.format( this.props ),
+		const currentFormat = this.props.value,
 			currentCursorPoint = this.numberInput.selectionStart,
-			nextFormat = this.format( nextProps );
+			nextFormat = nextProps.value;
 
 		let newCursorPoint = currentCursorPoint;
 		this.numberInput.value = nextFormat;
```

Comment:
That's a tricky one and the short answer is we can't. We need to have direct control over the exact steps to calculate the cursor's new position and move it appropriately.

The other option is to split this logic between `componentWillUpdate` and `componentDidUpdate` and store the cursor positions in component's state, but I think it makes everything more complex.